[DDP] Add more logging iterations

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #64073
* #64072
* __->__ #64071

Adding more logging iterations to get additional data.

Differential Revision: [D30579367](https://our.internmc.facebook.com/intern/diff/D30579367/)

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @agolynski @SciPioneer @H-Huang @mrzzd @cbalioglu @gcramer23

rohan-varma committed:
[DDP] Add more logging iterations

Adding more logging iterations to get additional data.

Differential Revision: [D30579367](https://our.internmc.facebook.com/intern/diff/D30579367/)

[ghstack-poisoned]

mrshenli reviewed (approved):
Stamp to unblock, but lint failure is real. Please fix before landing. 

```
/__w/pytorch/pytorch/torch/csrc/distributed/c10d/logger.cpp:11:7: error: do not declare C-style arrays, use std::array<> instead [cppcoreguidelines-avoid-c-arrays,modernize-avoid-c-arrays,-warnings-as-errors]
const int LoggingIterations[] = {10, 20, 100, 1000, 5000, 10000, 20000};
```

rohan-varma committed:
Update on "[DDP] Add more logging iterations"


Adding more logging iterations to get additional data.

Differential Revision: [D30579367](https://our.internmc.facebook.com/intern/diff/D30579367/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[DDP] Add more logging iterations"


Adding more logging iterations to get additional data.

Differential Revision: [D30579367](https://our.internmc.facebook.com/intern/diff/D30579367/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[DDP] Add more logging iterations"


Adding more logging iterations to get additional data.

Differential Revision: [D30579367](https://our.internmc.facebook.com/intern/diff/D30579367/)

[ghstack-poisoned]
